src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,7 +1625,16 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                if (!request.isSilent() && callbackHandle.callback != null && activity != null) {
+                    return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                } else {
+                    // cause exception added to indicate that refresh token was wiped
+                    callbackHandle.onError(new AuthenticationException(
```

Comment:
First, I think this change contains extra data that would be nice to have on client side (exception and server logs), and second it would be nice to see your change against this change, that way is easier to see that your refactoring doesn't break this logic